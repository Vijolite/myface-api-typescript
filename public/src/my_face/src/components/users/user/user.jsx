import React from "react";
import {Link} from 'react-router-dom';
import './user.scss';

export function User({ user }) {
//export function User() {

    // const user = {
    //     "id": 58,
    //     "name": "Anatollo Assinder",
    //     "username": "aassinder1l",
    //     "email": "aassinder1l@blinklist.com",
    //     "coverImageUrl": "https://picsum.photos/id/302/2100/800",
    //     "profileImageUrl": "https://robohash.org/aassinder1l.png?bgset=bg1",
    // }

    //<img src={user.coverImageUrl} alt={user.name} />
    return <div className="user">
        <div className="user_name">
            <Link to = {`/users/${user.id}`}>
                <h2>{user.name}</h2>
            </Link>
        </div>
        <div className="user_details">
            <p>Username: {user.username}</p>
            <p>Email:  {user.email}</p>
        </div>
        <div className="user_img">
            <Link  to = {`/users/${user.id}`}>
                <img src={user.profileImageUrl} alt={user.name} />
            </Link>
        </div>

    </div>
}