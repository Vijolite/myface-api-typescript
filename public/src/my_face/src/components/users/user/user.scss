@import "../../../constants.scss";

//background-color: $backgroundColor;

.user {
    background-color: $insideBackgroundColor;//$backgroundColor;
    color: $textColor;
    display: grid;
    grid-template-columns:  1fr 40vw;
    grid-template-rows:  0.6fr 0.4fr;
    height: 40vw;
    margin: 10px;
    border-color: $borderColor;
    border-style: solid;
    
    .user_name 
    {
        grid-column: 1/2;
        grid-row: 1/2;
        //align-self: center;
        font-size: 1em;
        //height: 50%;
        height: 20vw;
        a {
            color:rgb(241, 241, 177);//color:$textColor;
        }
    }
    .user_details {
        grid-column: 1/2;
        grid-row: 2/3;
        font-size:0.7em;
    }
    .user_img {
        grid-column: 2/3;
        grid-row: 1/3;
        width: 40vw;
        height: 40vw;
        align-self: center;
        margin: 0;
        //padding: 0;
        // width:100%;
        // height:100%;
        img {
            width: 40vw;
            height: 40vw;
        }

    }
    
}