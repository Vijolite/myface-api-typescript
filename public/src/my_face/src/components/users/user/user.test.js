import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { User } from "./user.jsx";
import {BrowserRouter} from 'react-router-dom';


test("renders message", () => {
  // Arrange
  const user = {
    id:1,
    name: "User1",
    username: "user name",
    email: "user@user",
    postprofileImageUrl: "User image 1",
  };

  // Act
  render(<BrowserRouter><User user={user} /></BrowserRouter>
    );
  const messageElement = screen.getByText(/user name/i);

  // render(<User user={user} />);
  // const messageElement = screen.getByText(/user name/i);

  // Assert
  expect(messageElement).toBeInTheDocument();
});

test("matches snapshot", () => {
  // Arrange
  const user = {
    id:1,
    name: "User1",
    username: "user name",
    email: "user@user",
    postprofileImageUrl: "User image 1",
  };

  // Act
  const tree = renderer
    .create(<BrowserRouter><User user={user} /></BrowserRouter>)
    .toJSON();

  // Assert
  expect(tree).toMatchSnapshot();
});

