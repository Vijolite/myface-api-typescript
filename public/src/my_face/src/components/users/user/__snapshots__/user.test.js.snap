// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="user"
>
  <div
    className="user_name"
  >
    <a
      href="/users/1"
      onClick={[Function]}
    >
      <h2>
        User1
      </h2>
    </a>
  </div>
  <div
    className="user_details"
  >
    <p>
      Username: 
      user name
    </p>
    <p>
      Email:  
      user@user
    </p>
  </div>
  <div
    className="user_img"
  >
    <a
      href="/users/1"
      onClick={[Function]}
    >
      <img
        alt="User1"
      />
    </a>
  </div>
</div>
`;
